src/index.js

```diff
@@ -8,3 +10,32 @@ export {
 
 export * from './service';
 export * from './message';
+
+export default (config) => {
+  const internalLogger = new Logger(config);
+
+  return (label) => {
+    const labelledLogger = internalLogger.Label(label);
+
+    const logger = level => (...args) => {
+      let msg;
+
+      if (args.length === 2) {
```

Comment:
second though, we can still expose the simple one-method API as you've suggested, but also check for `arg.length === 1 && arg[0] instanceof LogMessage` and skip the message creation
so we will be able to do something like 
```js
labelledLogger.warn(new PaymentErrorMessage('some payment anomaly happened', { additionalInfo } ));
```